java/src/jmri/util/usb/RailDriverMenuItem.java

```diff
@@ -0,0 +1,776 @@
+                buff[outIdx] = QUESTIONMARK;
+            } else if ((c >= 'A') && (c <= 'Z')) {
+                // Get seven segment code for alpha.
+                buff[outIdx] = SevenSegmentAlpha[c - 'A'];
+            } else if ((c >= 'a') && (c <= 'z')) {
+                // Get seven segment code for alpha.
+                buff[outIdx] = SevenSegmentAlpha[c - 'a'];
+            } else if (c == '-') {
+                buff[outIdx] = DASHSEGMENT;
+            } else // Is it a decimal point?
+            if (c == '.') {
+                // If so, OR in the decimal point segment.
+                buff[outIdx + 1] |= DPSEGMENT;
+                outIdx++;
+            } else {    // everything else is ignored
+                outIdx++;
+            }
+            outIdx--;
+            if (outIdx < 0) {
+                if (++i < ledstring.length()) {
+                    if (ledstring.charAt(i) == '.') {
+                        buff[0] |= DPSEGMENT;
+                    }
+                }
+                break;
+            }
+        }
+        sendMessage(buff, LEDCommand);
+    }   // setLEDs
+
+    public void setSpeakerOn(boolean onFlag) {
+        byte[] buff = new byte[7]; // data buffer
+        Arrays.fill(buff, (byte) 0);
+
+        buff[5] = (byte) (onFlag ? 1 : 0);      // On / off
+
+        sendMessage(buff, SpeakerCommand);
+    }   // setSpeakerOn
+
+    // Turn speaker on.
+    public void speakerOn() {
+        setSpeakerOn(true);
+    }
+
+    // Turn speaker off.
+    public void speakerOff() {
+        setSpeakerOn(false);
+    }
+
+    /**
+     * send message to hid device {p}
+     * <p>
+     * @param message   the message to send
+     * @param reportID  the report ID
+     */
+    private void sendMessage(byte[] message, byte reportID) {
+        // Ensure device is open after an attach/detach event
+        if (!hidDevice.isOpen()) {
+            hidDevice.open();
+        }
+
+        try {
+            int ret = hidDevice.write(message, message.length, reportID);
+            if (ret >= 0) {
+                log.debug("hidDevice.write returned: {}", ret);
+            } else {
+                log.error("hidDevice.write error: {}", hidDevice.getLastErrorMessage());
+            }
+        } catch (IllegalStateException ex) {
+            log.error("hidDevice.write Exception : {}", ex);
+        }
+    }
+
+    /*
+     * {@inheritDoc}
+     */
+    @Override
+    public void hidDeviceAttached(HidServicesEvent event) {
+        log.info("hidDeviceAttached({})", event);
+        HidDevice tHidDevice = event.getHidDevice();
+        if (tHidDevice.getVendorId() == VENDOR_ID) {
+            if (tHidDevice.getProductId() == PRODUCT_ID) {
+//                if ((SERIAL_NUMBER == null) || (tHidDevice.getSerialNumber().equals(SERIAL_NUMBER))) {
+                    if (!invokeOnMenuOnly) {
+                        setupRailDriver();
+                    }
+//                }
+            }
```

Comment:
I think (but I'm not entirely sure) that the problem on line 454 about useless control flow is due to that `if (!invokeOnMenuOnly) {` will never be true.